src/java.base/share/classes/com/sun/crypto/provider/CipherCore.java

```diff
@@ -1026,16 +1058,15 @@ private void endDoFinal() {
         }
     }
 
-    private int unpad(int outLen, byte[] outWithPadding)
+    private int unpad(int outLen, int off, byte[] outWithPadding)
```

Comment:
This appears to be a back-port of part of
* [8253821: Improve ByteBuffer performance with GCM](https://github.com/ibmruntimes/openj9-openjdk-jdk11/commit/d8fc6df82151f89566508fcb9da1e18f9a236527)

@KostasTsiounis Is that your understanding?

That seems reasonable, but should have been called out in the commit message and the description here.

Reply:
I didn't see this PR you are mentioning. I needed the extra functionality for my change to work and it wasn't available in OpenJDK11.

Should I change the description and/or the commit message. And if yes, what would the change be?

Reply:
The commit I mentioned above is derived from upstream; see also https://bugs.openjdk.org/browse/JDK-8253821 which says it was applied to Java 16. The commit message and the description should discuss that.

Reply:
In our jdk17 repo we have https://github.com/ibmruntimes/openj9-openjdk-jdk17/commit/d8fc6df82151

Reply:
The commit message is a little terse, but there's enough information here now.